aten/src/ATen/core/ivalue_inl.h

```diff
@@ -1023,19 +1032,20 @@ struct C10_EXPORT ivalue::Future final : c10::intrusive_ptr_target {
    * this function will execute the callback immediately.
    */
   template <typename T>
-  void addCallback(T callback) {
+  void addCallback(T callback, bool uses_future = true) {
 #if __cpp_lib_is_invocable >= 201703
     static_assert(
         std::is_invocable_r<void, T, Future&>::value,
-        "The callback must have signature void(Future&)");
+        "The callback must have signature void(Future&) or void()");
```

Comment:
is the `or void()` thing stale? i didn't see implementations for a void() callback.  (And if it is supported, then why not use it in the PGNCCL callbacks below?)

Reply:
you're right, it's stale - let me remove that. I was trying to get `addCallback` to take `void()` lambdas and I got it working on linux builds but not with windows builds